graylog2-server/src/main/java/org/graylog2/plugin/inputs/transports/GeneratorTransport.java

```diff
@@ -60,6 +62,16 @@ protected void run() throws Exception {
                     }
                 }
             }
+
+            @Override
+            protected void startUp() throws Exception {
+                runThread = Thread.currentThread();
+            }
+
+            @Override
+            protected void triggerShutdown() {
+                runThread.interrupt();
```

Comment:
Are there situations where `runThread` might be `null` because `triggerShutdown()` is called without calling `startUp` before`? If so, we should probably add a null check for `runThread` here.

Reply:
I think the service lifecycle guarantees prevent that from happening: https://guava.dev/releases/19.0/api/docs/com/google/common/util/concurrent/Service.html